services/gitlab.service

```diff
@@ -9,6 +9,7 @@ TimeoutStartSec=0
 TimeoutStopSec=15
 Restart=always
 RestartSec=5s
+ExecStartPre=/opt/bin/gitlab-network start
```

Comment:
No, totally unrelated (for http, it goes via docker DNS) - the crucial bit is that we restart HAProxy once the IP changes, which we do via the peridoic timer/pathchanged combo